### PR TITLE
adding support for index_select on quantized tensors

### PR DESCRIPTION
**Pull request by HDCharles**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#61406**

Summary:
Only really needed to fix a few select functions so that they could work
for quantized tensors. Primarily creation and resizing of tensors
required a branch for quantized tensors. This doesn't work for
per_channel tensors

Test Plan: python test/test_quantization.py
TestQuantizedTensor.test_qtensor_index_select_cuda

python test/test_quantization.py
TestQuantizedTensor.test_qteensor_index_select_cpu
Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D29654446](https://our.internmc.facebook.com/intern/diff/D29654446)

### CODE REVIEW

**Comment by HDCharles** [comment]
@HDCharles has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29654446).

**Comment by HDCharles** [comment]
This is intended as a fix for https://github.com/pytorch/pytorch/issues/59928

**Comment by HDCharles** [comment]
@HDCharles has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29654446).

File: test/quantization/core/test_quantized_tensor.py [hunk 1]
```diff
@@ -783,6 +783,32 @@ def test_qtensor_fill(self):
...
+            dims = 3
+            index = torch.randint(dims, [1]).item()
+            selected = torch.randperm(dims)[:2].to(device)
+            scale = 1
+            zp = 0
+
+            x = torch.randn([3] * dims, device=device) * 10
+            q = torch.quantize_per_tensor(x, scale, zp, quant_type)
+            qs = torch.index_select(q, index, selected)
+            for altered_index in range(2):
```
Comment (jerryzh168) [inline test/quantization/core/test_quantized_tensor.py:804]: should we just compare the following two path matches:
1. float tensor -> index_select -> quantize
2. float tensor -> quantize -> index_select

**Comment by HDCharles** [comment]
@HDCharles has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29654446).

**Comment by HDCharles** [comment]
@HDCharles has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29654446).

File: test/quantization/core/test_quantized_tensor.py (hunk 1 above)
Comment (HDCharles) [reply test/quantization/core/test_quantized_tensor.py:804]: oh yeah thats much better, fixed in newest version

**Comment by HDCharles** [comment]
@HDCharles has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29654446).

**Review by jerryzh168 — APPROVED** [review]
LG, thanks!

**Review by raghuramank100 — APPROVED** [review]

**Pull request closed without merging**